### PR TITLE
Support updating the database

### PR DESCRIPTION
**Pull request by myousfi96**
In this PR we have the following changes:

1. Fix duplicate entries in the Label field when updating the database.
2. Support tracking for what we need to update using the Timeline API from Github.
3. Break the extended function "parse_pr_list" into smaller pieces.

### CODE REVIEW

File: smartshark_plugin.py [hunk 1]
```diff
@@ -51,7 +51,6 @@ def main(args):
         pr_system = PullRequestSystem.objects.get(url=cfg.tracking_url, project_id=project.id)
     except PullRequestSystem.DoesNotExist:
         pr_system = PullRequestSystem(project_id=project.id, url=cfg.tracking_url)
-    pr_system.last_updated = datetime.datetime.now()
```
Comment (atrautsch) [inline smartshark_plugin.py:54]: good catch!

File: prSHARK/backends/github.py [hunk 2]
```diff
@@ -47,7 +48,8 @@ def _send_request(self, url):
             resp = requests.get(url, headers=headers, proxies=self.config.get_proxy_dictionary(), auth=auth)
 
             if resp.status_code != 200:
-                self._log.error("Problem with getting data via url %s. Code: %s, Error: %s", url, resp.status_code, resp.text)
```
Comment (atrautsch) [inline prSHARK/backends/github.py:50]: having only resp.text on a newline does not look more readable to me

File: prSHARK/backends/github.py [hunk 3]
```diff
@@ -56,11 +58,13 @@ def _send_request(self, url):
                     if r:
                         if 'errors' in r.keys():
                             for e in r['errors']:
-                                if e['resource'] == 'PullRequest' and e['field'] == 'diff' and e['code'] == 'not_available' and 'per_page' in url:  # we try to be as explicit as possible here
+                                if e['resource'] == 'PullRequest' and e['field'] == 'diff' and e[
```
Comment (atrautsch) [inline prSHARK/backends/github.py:61]: should maybe split on the and instead of the dict lookup

File: prSHARK/backends/github.py [hunk 4]
```diff
@@ -56,11 +58,13 @@ def _send_request(self, url):
...
+                                    'code'] == 'not_available' and 'per_page' in url:  # we try to be as explicit as possible here
                                     self._log.error('unfetchable files for pull request, returning [], try: %s', tries)
                                     return []
                 if resp.status_code == 500 and tries == 2:  # problem for some part of endpoints, e.g., https://api.github.com/repos/apache/kafka/pulls/3490/reviews/48104142/comments?&page=1&per_page=100
-                    self._log.error("Problem with getting data via url %s. Code: %s, Error: %s", url, resp.status_code, resp.text)
+                    self._log.error("Problem with getting data via url %s. Code: %s, Error: %s", url, resp.status_code,
```
Comment (atrautsch) [inline prSHARK/backends/github.py:66]: having only resp.text on a newline does not look more readable to me

File: prSHARK/backends/github.py [hunk 5]
```diff
@@ -220,9 +223,20 @@ def fetch_event_list(self, pr_number):
...
+        """
+        base_url = self.config.tracking_url.replace('/pulls', '/issues')
+        url = '{}/{}/timeline?'.format(base_url, pr_number)
+        return self._fetch_all_pages(url)
+
     def fetch_pr_list(self):
         """Fetch complete list of pull requests for this the url passed on the command line."""
-        url = '{}?state=all'.format(self.config.tracking_url)  # this is where we would put since=last_updated_at if it would be supported by the github api
+        # this is where we would put since=last_updated_at if it would be supported by the github api
+        url = '{}?state=all'.format(
```
Comment (atrautsch) [inline prSHARK/backends/github.py:238]: could be all in one line

**Review by atrautsch — CHANGES_REQUESTED** [review]
Good changes overall, I have some nits for the style changes.
There are tests and fixtures for this repository. Did the tests run after refactoring?
If yes, you should also extend the existing tests and fixtures with the timeline changes.

**Comment by myousfi96** [comment]
Thank you for the review. I've included a test for the timeline and tested it. Regarding the style changes, you're correct, I didn't intend to make them. It's just a habit of mine to accidentally press a shortcut that reformats the entire file

**Review by atrautsch — APPROVED** [review]

**Pull request merged**